src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -104,84 +51,21 @@ long alignof(List<MemoryLayout> elems) {
      *
      * @return the member layouts associated with this group.
      */
-    public List<MemoryLayout> memberLayouts() {
-        return Collections.unmodifiableList(elements);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public String toString() {
-        return decorateLayoutString(elements.stream()
-                .map(Object::toString)
-                .collect(Collectors.joining(kind.delimTag, "[", "]")));
-    }
+    List<MemoryLayout> memberLayouts();
 
     /**
      * {@return {@code true}, if this group layout is a struct layout}
      */
-    public boolean isStruct() {
-        return kind == Kind.STRUCT;
-    }
+    boolean isStruct();
```

Comment:
I totally agree. This also applies to `isPadding()`. I was a bit unsure whether or not this change should be in this PR or a separate one. Since it is changing the API, my initial thought was to introduce this in a separate PR.  What is your opinion on this?

Reply:
I meant to say "Perhaps a change to consider *after* this refactoring" i.e. lets consider such changes in subsequent PRs.